core/application/src/app.rs

```diff
@@ -100,32 +99,16 @@ impl<C: NodeComponents> ApplicationInterface<C> for Application<C> {
         checkpoint: Vec<u8>,
         checkpoint_hash: [u8; 32],
     ) -> Result<()> {
-        // Due to a race condition on shutdowns when a node checkpoints, we should sleep and try
```

Comment:
I wasn't able to reproduce this anymore after running the test in a loop non-stop for a while. I was able to reproduce the rocks "`LOCK` still held by current process" error from here if there is no wait_for_ready or sleep after startup before shutdown, but this loop doesn't solve that, it just loops and fails at the end.

Reply:
but we call wait for ready which solves it right?